src/server/session.ts

```diff
@@ -459,70 +489,69 @@ namespace ts.server {
-                const start = compilerService.host.positionToLineOffset(ref.fileName, ref.textSpan.start);
-                const refLineSpan = compilerService.host.lineToTextSpan(ref.fileName, start.line - 1);
-                const snap = compilerService.host.getScriptSnapshot(ref.fileName);
-                const lineText = snap.getText(refLineSpan.start, ts.textSpanEnd(refLineSpan)).replace(/\r|\n/g, "");
-                return {
-                    file: ref.fileName,
-                    start: start,
-                    lineText: lineText,
-                    end: compilerService.host.positionToLineOffset(ref.fileName, ts.textSpanEnd(ref.textSpan)),
-                    isWriteAccess: ref.isWriteAccess
-                };
-            }).sort(compareFileStart);
+            const nameColStart = defaultProject.compilerService.host.positionToLineOffset(file, nameSpan.start).offset;
+            const nameText = defaultProject.compilerService.host.getScriptSnapshot(file).getText(nameSpan.start, ts.textSpanEnd(nameSpan));
+            const refs = forEachProject<protocol.ReferencesResponseItem>(
+                projects,
+                (project: Project) => {
+                    const compilerService = project.compilerService;
+                    const references = compilerService.languageService.getReferencesAtPosition(file, position);
+                    if (!references) {
+                        return [];
+                    }
+
+                    return references.map(ref => {
+                        const start = compilerService.host.positionToLineOffset(ref.fileName, ref.textSpan.start);
+                        const refLineSpan = compilerService.host.lineToTextSpan(ref.fileName, start.line - 1);
+                        const snap = compilerService.host.getScriptSnapshot(ref.fileName);
+                        const lineText = snap.getText(refLineSpan.start, ts.textSpanEnd(refLineSpan)).replace(/\r|\n/g, "");
+                        return {
+                            file: ref.fileName,
+                            start: start,
+                            lineText: lineText,
+                            end: compilerService.host.positionToLineOffset(ref.fileName, ts.textSpanEnd(ref.textSpan)),
+                            isWriteAccess: ref.isWriteAccess
+                        };
+                    });
+                },
+                compareFileStart
+            );
+
             return {
-                refs: bakedRefs,
+                refs: deduplicate(refs, areReferencesResponseItemsForTheSameLocation),
                 symbolName: nameText,
                 symbolStartOffset: nameColStart,
                 symbolDisplayString: displayString
             };
+
+            function areReferencesResponseItemsForTheSameLocation(a: protocol.ReferencesResponseItem, b: protocol.ReferencesResponseItem) {
+                if (a && b) {

```

Comment:
Originally it might be undefined when `getReferencesAtPosition` returns undefined. Now it is not likely, though this is just to be conservative.